scripts: cli: Add option allowing the user to provide an external DTS overlay file

alla3 opened the pull request:
Add option `--dts_ovl_path` on `make` command allowing the user to provide an external DTS overlay file path and check for it in the application folder whenever the user does not provide an external path. If the file does not exist, the system will use the default DTS files.

It is not necessary to provide a path in the command line if the file is placed on the root application folder and it is named as `<BOARD>-overlay.dts`.

Closes #62.

alla3 committed:
scripts: cli: Add option to pass a DTS overlay

Add option '--dts_ovl_path' on 'make' command allowing the user to
provide an external DTS overlay file path.

Signed-off-by: Allysson Lukas <alla@cesar.org.br>

ramonhpr commented on scripts/cli.py:
```diff
@@ -247,6 +248,22 @@ def set_ext_proto_path(self, proto_path=None):
+            self.ext_dts_ovl_path = dts_ovl_path
+
+        # Log in case of a DTS overlay path passed
+        if self.ext_dts_ovl_path is not None:
```
The only place where you set `self.ext_dts_ovl_path` is in the if above.
I think that this if's can be merged

ramonhpr replied:
This if just makes sense on next commit

alla3 replied:
Done.

ramonhpr commented on scripts/cli.py:
```diff
@@ -152,6 +152,12 @@ class Constants:
         BOARD_ALIASES = {'dk':     BOARD_DK,
                          'mesh':   BOARD_MESH,
                          'dongle': BOARD_DONGLE}
+        USER_DTS_OVERLAY_FILE = {BOARD_DK:     'nrf52840_pca10056-overlay.dts',
```
You can do a patch improving this. Make something like that:
```
BOARDS = [BOARD_DK, BOARD_MESH, BOARD_DONGLE]
USER_DTS_OVERLAY_FILE =  {board: board+'-overlay.dts' for board in BOARDS }
USER_DTS_OVERLAY_FILE_ALIASES = {board: board+'-overlay.dts' for board in BOARD_ALIASES }
```
This will help to change just the array with the supported boards :)

alla3 replied:
Done.

alla3 committed:
scripts: cli: Check for a local DTS overlay file

Check for a DTS overlay file in the application folder whenever the user
does not provide an external path using the option '--dts_ovl_path' on
'make' command.

The overlay file should be named as <BOARD>.overlay and it should
be placed on the root application folder.

If the file does not exist, the system will use the default DTS files.

Signed-off-by: Allysson Lukas <alla@cesar.org.br>

alla3 committed:
scripts: cli: Check for an alternative DTS overlay

Check for a DTS overlay file named as '<BOARD_ALIASES>.overlay' in the
main application directory whenever the user does not provide an
external path using the option '--dts_ovl_path' on 'make' command and
the '<BOARD>.overlay' file is not found.

If the file does not exist, the system will use the default DTS files.

Signed-off-by: Allysson Lukas <alla@cesar.org.br>
